### PR TITLE
Skel tweaks

### PR DESCRIPTION
**Pull request by jstypka**
Doing some modification to the skel version in order to improve its performance

### CODE REVIEW

File: src/skel/mas_skel.erl [hunk 1]
```diff
@@ -42,7 +43,6 @@ main(Population, Time, SP, Cf) ->
     Workers = Cf#config.skel_workers,
 
     TagFun = fun({Home, Agent}) ->
-                     seed_random_once_per_process(),
                      {{Home, mas_misc_util:behaviour_proxy(Agent, SP, Cf)}, Agent}
```
Comment (elvisci) [inline src/skel/mas_skel.erl]: Line 46 is too long:                      {{Home, mas_misc_util:behaviour_proxy(Agent, SP, Cf)}, Agent}.

File: src/skel/skel_fast.erl [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+%% ====================================================================
+%% API functions
+%% ====================================================================
+-spec start(Time::pos_integer(), sim_params(), config()) -> ok.
+start(Time, SP, Cf = #config{islands = Islands, agent_env = Env}) ->
+    mas_topology:start_link(self(), Islands, Cf#config.topology),
+    mas_misc_util:seed_random(),
+    mas_misc_util:clear_inbox(),
+    ets:new(migration_ets, [duplicate_bag, public, named_table]),
+    Population = [{I, [ Env:initial_agent(SP)
```
Comment (elvisci) [inline src/skel/skel_fast.erl]: Remove the dynamic function call on line 17. Only modules that define callbacks should make dynamic calls.

File: src/skel/skel_fast.erl [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+                      [true = ets:delete_object(migration_ets, Imm)
+                       || Imm <- Immigrants],
+
+                      Im = [A || {_Island, A} <- Immigrants],
+
+                      case Im ++ Emigrants of
+                          [] -> nothing;
+                          _ ->
+                              io:format("Island ~p - emigrated ~p, immigrated ~p, leftovers ~p~n",
```
Comment (elvisci) [inline src/skel/skel_fast.erl]: Line 80 is too long:                               io:format("Island ~p - emigrated ~p, immigrated ~p, leftovers ~p~n",.

Comment (jstypka) [inline src/skel/skel_fast.erl]: This is just debugging code that has been accidentally committed. Will be eventually removed.

File: src/skel/skel_fast.erl [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+-spec start(Time::pos_integer(), sim_params(), config()) -> [mas:agent()].
+start(Time, SP, Cf = #config{islands = Islands, agent_env = Env}) ->
+    mas_topology:start_link(self(), Islands, Cf#config.topology),
+    mas_misc_util:seed_random(),
+    mas_misc_util:clear_inbox(),
+    ets:new(migration_ets, [duplicate_bag,
+                            public,
+                            named_table,
+                            {read_concurrency, true}]),
+    Population = [{I, [ Env:initial_agent(SP)
```
Comment (elvisci) [inline src/skel/skel_fast.erl]: Remove the dynamic function call on line 20. Only modules that define callbacks should make dynamic calls.

File: src/skel/mas_skel.erl [hunk 5]
```diff
@@ -1,33 +1,30 @@
...
     mas_misc_util:seed_random(),
     mas_misc_util:clear_inbox(),
-    Population = [{I, Env:initial_agent(SP)} ||
-                     _ <- lists:seq(1, Cf#config.population_size),
-                     I <- lists:seq(1, Islands)],
+    ets:new(migration_ets, [duplicate_bag,
+                            public,
+                            named_table,
+                            {read_concurrency, true}]),
+    Population = [{I, [ Env:initial_agent(SP)
```
Comment (elvisci) [inline src/skel/mas_skel.erl]: Remove the dynamic function call on line 20. Only modules that define callbacks should make dynamic calls.

File: src/skel/mas_skel_old.erl [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+%% ====================================================================
+%% API functions
+%% ====================================================================
+
+-spec start(Time::pos_integer(), sim_params(), config()) -> [mas:agent()].
+start(Time, SP, Cf = #config{islands = Islands, agent_env = Env}) ->
+        mas_topology:start_link(self(), Islands, Cf#config.topology),
+        mas_misc_util:seed_random(),
+        mas_misc_util:clear_inbox(),
+        Population = [{I, Env:initial_agent(SP)} ||
```
Comment (elvisci) [inline src/skel/mas_skel_old.erl]: Remove the dynamic function call on line 24. Only modules that define callbacks should make dynamic calls.

File: src/skel/mas_skel_old.erl [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+%% @doc Main program loop
+-spec main([{pos_integer(), mas:agent()}],
+           non_neg_integer(), sim_params(), config()) -> [mas:agent()].
+main(Population, Time, SP, Cf) ->
+    EndTime = mas_misc_util:add_miliseconds(os:timestamp(), Time),
+    Workers = Cf#config.skel_workers,
+
+    TagFun = fun({Home, Agent}) ->
+                     {{Home, mas_misc_util:behaviour_proxy(Agent, SP, Cf)}, Agent}
```
Comment (elvisci) [inline src/skel/mas_skel_old.erl]: Line 45 is too long:                      {{Home, mas_misc_util:behaviour_proxy(Agent, SP, Cf)}, Agent}.

File: src/skel/mas_skel_old.erl [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+             end,
+
+    MigrateFun = fun({{Home, migration}, Agent}) ->
+                         {{mas_topology:getDestination(Home), migration}, Agent};
```
Comment (elvisci) [inline src/skel/mas_skel_old.erl]: Line 49 is too long:                          {{mas_topology:getDestination(Home), migration}, Agent};.

File: src/skel/mas_skel_old.erl [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+%% ====================================================================
+
+-spec log_countstats([tuple()], config()) -> ok.
+log_countstats(Groups, Cf) ->
+    BigDict = dict:from_list([{I, mas_misc_util:create_new_counter(Cf)}
+                              || I <- lists:seq(1, Cf#config.islands)]),
+
+    NewBigDict = lists:foldl(fun({{Home, Behaviour}, Group}, AccBD) ->
+                                     IslandDict = dict:fetch(Home, AccBD),
+                                     NewIslandDict = dict:update_counter(Behaviour, length(Group), IslandDict),
```
Comment (elvisci) [inline src/skel/mas_skel_old.erl]: Line 122 is too long:                                      NewIslandDict = dict:update_counter(Behaviour, length(Group), IslandDict),.

**Comment by jstypka** [comment]
Closed, because another identical pull request have been opened on an updated master branch

**Pull request closed without merging**